coordinator/src/node.rs

```diff
@@ -116,10 +146,21 @@ impl Node {
 
         let closing_price = trade_params.average_execution_price();
 
-        // FIXME: This is wrong as we cannot use the closing price to calculated the
-        // `accept_settlement_amount`. We must save the initial price when creating the position and
-        // use it here again for closing.
-        let opening_price = closing_price;
+        let opening_price = match self
+            .positions
+            .lock()
+            .expect("to get lock on positions")
+            .remove(&trade_params.pubkey.to_string())
+        {
+            Some(position) => Decimal::try_from(position.average_entry_price)?,
+            None => {
+                tracing::warn!("Did not find position in memory, thus we do not have the opening price to calculate a correct accept settlement amount. Using the closing price.");
+                // FIXME: This is wrong as we cannot use the closing price to calculated the
+                // `accept_settlement_amount`. We must save the initial price when creating the
+                // position and use it here again for closing.
+                closing_price
+            }
```

Comment:
👍 And we know this will happen if the coordinator is restarted, since the position info won't be persisted.